### ISSUE TITLE
Error running test_syphilis on refactor-rngs-7

### ISSUE THREAD

**Issue by robynstuart** [starsimhub/starsim#409]
Possibly due to Python 3.9?

```
Traceback (most recent call last):
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3526, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-a4103bcf5407>", line 1, in <module>
    runfile('/Users/robynstuart/Documents/git/starsim/tests/test_syphilis.py', wdir='/Users/robynstuart/Documents/git/starsim')
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/_pydev_bundle/pydev_umd.py", line 198, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/Users/robynstuart/Documents/git/starsim/tests/test_syphilis.py", line 183, in <module>
    sim = test_syph(dt=dt, n_agents=n_agents)
  File "/Users/robynstuart/Documents/git/starsim/tests/test_syphilis.py", line 94, in test_syph
    sim = ss.Sim(analyzers=check_states(), **sim_kwargs)
  File "/Users/robynstuart/Documents/git/starsim/starsim/sim.py", line 52, in __init__
    self.dists = ss.Dists(obj=self)
  File "/Users/robynstuart/Documents/git/starsim/starsim/distributions.py", line 63, in __init__
    self.initialize()
  File "/Users/robynstuart/Documents/git/starsim/starsim/distributions.py", line 82, in initialize
    dist.initialize(trace=trace, seed=base_seed, sim=sim)
  File "/Users/robynstuart/Documents/git/starsim/starsim/distributions.py", line 263, in initialize
    self.set_offset(trace, seed)
  File "/Users/robynstuart/Documents/git/starsim/starsim/distributions.py", line 288, in set_offset
    self.offset = str2int(unique_name) # Key step: hash the path to the distribution
  File "/Users/robynstuart/Documents/git/starsim/starsim/distributions.py", line 19, in str2int
    return int.from_bytes(string.encode()) % modulo
TypeError: from_bytes() missing required argument 'byteorder' (pos 2)
```

**Comment by daniel-klein** [comment]
@cliffckerr, I fixed this in #411. Just need to a byteorder argument:
```python
return int.from_bytes(string.encode(), 'big') % modulo
```

**Comment by cliffckerr** [comment]
Looks like it changed to have a default value in Python 3.11:
> Changed in version 3.11: Added default argument values for length and byteorder.

Do we want to support Python 3.9 - 3.10? I guess we might as well...

**Comment by cliffckerr** [comment]
Fixed in #420
